### ISSUE TITLE
UpdateSql Command Step throws LockException on an empty database

### ISSUE THREAD

**Issue by filipelautert** [liquibase/liquibase#4150]
### Search first

- [X] I searched and no similar issues were found

### Description

* It fails only with UpdateSql command 
* the database is empty - if you run `update `and then `updateSQL `the problem goes away

What is happening: 
* create databasechangeloglock and insert command scripts are sent to stdout  (as expected?)
* the release lock script is being executed (and fails as the table wasn't created)

### Steps To Reproduce

```
    def "run UpdateSql"() {
        when:
        def jdbcURL = "jdbc:h2:file:~/test-h2-1";
        def connection =  DriverManager.getConnection(jdbcURL);
        def database = new H2Database()
        database.setConnection(new JdbcConnection(connection))
        def updateCommand = new CommandScope(UpdateSqlCommandStep.COMMAND_NAME)
        updateCommand.addArgumentValue(DbUrlConnectionCommandStep.DATABASE_ARG, database)
        updateCommand.addArgumentValue(UpdateSqlCommandStep.CHANGELOG_FILE_ARG, "liquibase/simple.yml")

        then:
        def result = updateCommand.execute().getResults()
    }
```
simple.yml can be any changelog.

Output:

```
OpenJDK 64-Bit Server VM warning: Ignoring option --illegal-access=warn; support was removed in 17.0
abr. 18, 2023 2:19:35 PM liquibase.database
INFO: Set default schema name to PUBLIC
abr. 18, 2023 2:19:35 PM liquibase.executor
INFO: Constructed an ExampleExecutor
-- Create Database Lock Table
abr. 18, 2023 2:19:35 PM liquibase.lockservice
INFO: Successfully acquired change log lock
abr. 18, 2023 2:19:36 PM liquibase.changelog
INFO: Creating database history table with name: PUBLIC.DATABASECHANGELOG
CREATE TABLE PUBLIC.DATABASECHANGELOGLOCK (ID INT NOT NULL, LOCKED BOOLEAN NOT NULL, LOCKGRANTED TIMESTAMP, LOCKEDBY VARCHAR(255), CONSTRAINT PK_DATABASECHANGELOGLOCK PRIMARY KEY (ID));

-- Initialize Database Lock Table
DELETE FROM PUBLIC.DATABASECHANGELOGLOCK;

INSERT INTO PUBLIC.DATABASECHANGELOGLOCK (ID, LOCKED) VALUES (1, FALSE);

-- Lock Database
UPDATE PUBLIC.DATABASECHANGELOGLOCK SET LOCKED = TRUE, LOCKEDBY = 'caelum (172.25.2.86)', LOCKGRANTED = NOW() WHERE ID = 1 AND LOCKED = FALSE;

-- Create Database Change Log Table
abr. 18, 2023 2:19:36 PM liquibase.command
INFO: Using deploymentId: 1838376065
abr. 18, 2023 2:19:36 PM liquibase.changelog
INFO: Table example_table created
abr. 18, 2023 2:19:36 PM liquibase.changelog
INFO: ChangeSet liquibase/simple.yml::1::Liquibase User ran successfully in 5ms
CREATE TABLE PUBLIC.DATABASECHANGELOG (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED TIMESTAMP NOT NULL, ORDEREXECUTED INT NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35), DESCRIPTION VARCHAR(255), COMMENTS VARCHAR(255), TAG VARCHAR(255), LIQUIBASE VARCHAR(20), CONTEXTS VARCHAR(255), LABELS VARCHAR(255), DEPLOYMENT_ID VARCHAR(10));

-- *********************************************************************
-- Update Database Script
-- *********************************************************************
-- Change Log: liquibase/simple.yml
-- Ran at: 18/04/2023 14:19
-- Against: @jdbc:h2:file:~/test-h2-2
-- Liquibase version: [local build]
-- *********************************************************************

-- Changeset liquibase/simple.yml::1::Liquibase User
CREATE TABLE PUBLIC.example_table (id INT GENERATED BY DEFAULT AS IDENTITY NOT NULL, firstname VARCHAR(50), surname VARCHAR(50) NOT NULL, state CHAR(2), CONSTRAINT PK_EXAMPLE_TABLE PRIMARY KEY (id));

abr. 18, 2023 2:19:36 PM liquibase.command
INFO: Update command completed successfully.
abr. 18, 2023 2:19:36 PM liquibase.lockservice
INFO: Successfully released change log lock
abr. 18, 2023 2:19:36 PM liquibase.lockservice
INFO: Successfully released change log lock
abr. 18, 2023 2:19:36 PM liquibase.command
SEVERE: Could not release lock
liquibase.exception.LockException: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: Error executing SQL UPDATE PUBLIC.DATABASECHANGELOGLOCK SET LOCKED = FALSE, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1: Table "DATABASECHANGELOGLOCK" not found (this database is empty); SQL statement:
UPDATE PUBLIC.DATABASECHANGELOGLOCK SET LOCKED = FALSE, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1 [42104-214]
	at liquibase.lockservice.StandardLockService.releaseLock(StandardLockService.java:397)
	at liquibase.command.core.helpers.LockServiceCommandStep.cleanUp(LockServiceCommandStep.java:50)
	at liquibase.command.CommandScope.execute(CommandScope.java:225)
	at liquibase.command.CommandScope$execute$0.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:120)
	at liquibase.command.UpdateSqlCommandTest.$spock_feature_0_0(UpdateSqlCommandTest.groovy:25)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.spockframework.util.ReflectionUtil.invokeMethod(ReflectionUtil.java:200)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:113)
	at org.spockframework.runtime.BaseSpecRunner.invokeRaw(BaseSpecRunner.java:484)
	at org.spockframework.runtime.BaseSpecRunner.invoke(BaseSpecRunner.java:467)
	at org.spockframework.runtime.BaseSpecRunner.runFeatureMethod(BaseSpecRunner.java:408)
	at org.spockframework.runtime.BaseSpecRunner.doRunIteration(BaseSpecRunner.java:322)
	at org.spockframework.runtime.BaseSpecRunner$6.invoke(BaseSpecRunner.java:306)
	at org.spockframework.runtime.BaseSpecRunner.invokeRaw(BaseSpecRunner.java:484)
	at org.spockframework.runtime.BaseSpecRunner.invoke(BaseSpecRunner.java:467)
	at org.spockframework.runtime.BaseSpecRunner.runIteration(BaseSpecRunner.java:285)
	at org.spockframework.runtime.BaseSpecRunner.initializeAndRunIteration(BaseSpecRunner.java:275)
	at org.spockframework.runtime.BaseSpecRunner.runSimpleFeature(BaseSpecRunner.java:266)
	at org.spockframework.runtime.BaseSpecRunner.doRunFeature(BaseSpecRunner.java:260)
	at org.spockframework.runtime.BaseSpecRunner$5.invoke(BaseSpecRunner.java:243)
	at org.spockframework.runtime.BaseSpecRunner.invokeRaw(BaseSpecRunner.java:484)
	at org.spockframework.runtime.BaseSpecRunner.invoke(BaseSpecRunner.java:467)
	at org.spockframework.runtime.BaseSpecRunner.runFeature(BaseSpecRunner.java:235)
	at org.spockframework.runtime.BaseSpecRunner.runFeatures(BaseSpecRunner.java:185)
	at org.spockframework.runtime.BaseSpecRunner.doRunSpec(BaseSpecRunner.java:95)
	at org.spockframework.runtime.BaseSpecRunner$1.invoke(BaseSpecRunner.java:81)
	at org.spockframework.runtime.BaseSpecRunner.invokeRaw(BaseSpecRunner.java:484)
	at org.spockframework.runtime.BaseSpecRunner.invoke(BaseSpecRunner.java:467)
	at org.spockframework.runtime.BaseSpecRunner.runSpec(BaseSpecRunner.java:73)
	at org.spockframework.runtime.BaseSpecRunner.run(BaseSpecRunner.java:64)
	at org.spockframework.runtime.Sputnik.run(Sputnik.java:63)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:232)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:55)
Caused by: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: Error executing SQL UPDATE PUBLIC.DATABASECHANGELOGLOCK SET LOCKED = FALSE, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1: Table "DATABASECHANGELOGLOCK" not found (this database is empty); SQL statement:
UPDATE PUBLIC.DATABASECHANGELOGLOCK SET LOCKED = FALSE, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1 [42104-214]
	at liquibase.executor.jvm.ChangelogJdbcMdcListener.query(ChangelogJdbcMdcListener.java:60)
	at liquibase.lockservice.StandardLockService.releaseLock(StandardLockService.java:363)
	... 41 more
Caused by: liquibase.exception.DatabaseException: Error executing SQL UPDATE PUBLIC.DATABASECHANGELOGLOCK SET LOCKED = FALSE, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1: Table "DATABASECHANGELOGLOCK" not found (this database is empty); SQL statement:
UPDATE PUBLIC.DATABASECHANGELOGLOCK SET LOCKED = FALSE, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1 [42104-214]
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:89)
	at liquibase.executor.jvm.JdbcExecutor.update(JdbcExecutor.java:311)
	at liquibase.executor.jvm.JdbcExecutor.update(JdbcExecutor.java:279)
	at liquibase.lockservice.StandardLockService.lambda$releaseLock$6(StandardLockService.java:363)
	at liquibase.executor.jvm.ChangelogJdbcMdcListener.lambda$query$1(ChangelogJdbcMdcListener.java:54)
	at liquibase.Scope.child(Scope.java:203)
	at liquibase.Scope.child(Scope.java:179)
	at liquibase.executor.jvm.ChangelogJdbcMdcListener.query(ChangelogJdbcMdcListener.java:54)
	... 42 more
Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "DATABASECHANGELOGLOCK" not found (this database is empty); SQL statement:
UPDATE PUBLIC.DATABASECHANGELOGLOCK SET LOCKED = FALSE, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1 [42104-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.command.Parser.getTableOrViewNotFoundDbException(Parser.java:8385)
	at org.h2.command.Parser.getTableOrViewNotFoundDbException(Parser.java:8379)
	at org.h2.command.Parser.getTableOrViewNotFoundDbException(Parser.java:8363)
	at org.h2.command.Parser.readTableOrView(Parser.java:8358)
INSERT INTO PUBLIC.DATABASECHANGELOG (ID, AUTHOR, FILENAME, DATEEXECUTED, ORDEREXECUTED, MD5SUM, DESCRIPTION, COMMENTS, EXECTYPE, CONTEXTS, LABELS, LIQUIBASE, DEPLOYMENT_ID) VALUES ('1', 'Liquibase User', 'liquibase/simple.yml', NOW(), 1, '8:7279a1672da799eb1c8b1618504f84a3', 'createTable tableName=example_table', '', 'EXECUTED', NULL, 'setup', 'DEV', '1838376065');
	at org.h2.command.Parser.readTableOrView(Parser.java:8328)

	at org.h2.command.Parser.readSimpleTableFilter(Parser.java:1160)
	at org.h2.command.Parser.parseUpdate(Parser.java:1118)
	at org.h2.command.Parser.parsePrepared(Parser.java:858)
-- Release Database Lock
	at org.h2.command.Parser.parse(Parser.java:689)
	at org.h2.command.Parser.parse(Parser.java:661)
UPDATE PUBLIC.DATABASECHANGELOGLOCK SET LOCKED = FALSE, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1;

	at org.h2.command.Parser.prepareCommand(Parser.java:569)
	at org.h2.engine.SessionLocal.prepareLocal(SessionLocal.java:631)
	at org.h2.engine.SessionLocal.prepareCommand(SessionLocal.java:554)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1116)
	at org.h2.jdbc.JdbcStatement.executeUpdateInternal(JdbcStatement.java:186)
	at org.h2.jdbc.JdbcStatement.executeUpdate(JdbcStatement.java:143)
	at liquibase.executor.jvm.JdbcExecutor$1UpdateStatementCallback.doInStatement(JdbcExecutor.java:302)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:77)
	... 49 more

abr. 18, 2023 2:19:36 PM liquibase.command
INFO: Command execution complete

Process finished with exit code 0

```

### Expected/Desired Behavior

Clean run.

### Liquibase Version

4.21.0+

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR (Thank you!)

**Comment by dominiquems** [comment]
Can confirm I'm also facing this issue using `liquibase-core` dependency on using Java 11. Deprecated `update` methods are still working, but non-deprecated ones do not lead to invocation of `CreateDatabaseChangeLogLockTableGenerator`'s `generateSql` method and thus do not create the databasechangeloglock table.

**Comment by dominiquems** [comment]
Hi @filipelautert, I tested with the jars from build https://github.com/liquibase/liquibase/actions/runs/4737226483?pr=4151 and the script to execute the release lock script is now executed on an empty database. 

I noticed that when using the `update(new Contexts(), new PrintWriter(System.out))`, the actual table(s) are not created in an empty database. This was not the case when using any of the deprecated `update` methods. 

[liquibase-log.txt](https://github.com/liquibase/liquibase/files/11271112/liquibase-log.txt)
